eternalcore-core/src/main/java/com/eternalcode/core/feature/jail/JailServiceImpl.java

```diff
@@ -0,0 +1,173 @@
+package com.eternalcode.core.feature.jail;
+
+import com.eternalcode.commons.bukkit.position.Position;
+import com.eternalcode.commons.bukkit.position.PositionAdapter;
+import com.eternalcode.core.configuration.ConfigurationManager;
+import com.eternalcode.core.configuration.implementation.LocationsConfiguration;
+import com.eternalcode.core.event.EventCaller;
+import com.eternalcode.core.feature.jail.event.JailDetainEvent;
+import com.eternalcode.core.feature.jail.event.JailReleaseEvent;
+import com.eternalcode.core.feature.spawn.SpawnService;
+import com.eternalcode.core.feature.teleport.TeleportService;
+import com.eternalcode.core.injector.annotations.Inject;
+import com.eternalcode.core.injector.annotations.component.Service;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.UUID;
+import javax.annotation.Nullable;
+import org.bukkit.Location;
+import org.bukkit.Server;
+import org.bukkit.command.CommandSender;
+import org.bukkit.entity.Player;
+import org.jetbrains.annotations.Blocking;
+
+import java.util.Optional;
+
+@Service
+class JailServiceImpl implements JailService {
+
+    private final Map<UUID, JailedPlayer> jailedPlayers = new HashMap<>();
+
+    private final TeleportService teleportService;
+    private final SpawnService spawnService;
+    private final JailSettings settings;
+    private final EventCaller eventCaller;
+    private final Server server;
+
+    private final PrisonerRepository prisonerRepository;
+    private final LocationsConfiguration locationsConfiguration;
+    private final ConfigurationManager configurationManager;
+
+    @Inject
+    JailServiceImpl(TeleportService teleportService, SpawnService spawnService, JailSettings settings, EventCaller eventCaller, Server server, PrisonerRepository prisonerRepository, LocationsConfiguration locationsConfiguration, ConfigurationManager configurationManager) {
```

Comment:
Use someone like this, correct my code.

```suggestion
    JailServiceImpl(
    TeleportService teleportService, 
    SpawnService spawnService, 
    JailSettings settings, 
    EventCaller eventCaller, 
    Server server, 
    PrisonerRepository prisonerRepository, 
    LocationsConfiguration locationsConfiguration, 
    ConfigurationManager configurationManager
    ) {
```